### PR TITLE
bpo-43725: Add CI step to check changes in the exported ABI

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue43725

### CODE REVIEW

**Comment by pablogsal** [comment]
Unfortunately seems that doing this check with two different compiler versions produce some different DWARF and the tool identifies the differences as errors :(

**Review by gpshead — APPROVED** [review]
Neat, lets try it out!  I suppose this shows up as a non-blocking check by default?  Release branch owners can presumably decide if this is reliable enough to make it blocking.

**Comment by pablogsal** [comment]
I will try to investigate how to overcome this:


> Unfortunately seems that doing this check with two different compiler versions produce some different DWARF and the tool identifies the differences as errors :(

**Comment by pablogsal** [comment]
@gpshead Check my message in https://bugs.python.org/msg390213 with the analysis of this solution.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -48,6 +48,28 @@ jobs:
             git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qvE '(\.rst$|^Doc|^Misc)' && echo '::set-output name=run_tests::true' || true
           fi
 
+  check_abi:
+    name: 'Check if the ABI has changed'
+    runs-on: ubuntu-latest
```
Comment (erlend-aasland) [inline .github/workflows/build.yml:53]: I'd pin the runner, to ensure that the job runs with at least the same major GCC version every time. Ref. bpo-43382.

**Pull request closed without merging**